libs/src/utils/multiProvider.js

```diff
@@ -0,0 +1,44 @@
+const { promisify, callbackify } = require('util')
+const Web3 = require('../web3')
+const { shuffle } = require('lodash')
+
+class MultiProvider extends Web3.providers.HttpProvider {
+  // web3 consumes a provider object on initialization
+  // ref: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3/types/index.d.ts#L31
+  // which references: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-core/types/index.d.ts#L436
+  // MultiProivder implements AbstractProvider which can be consumed by web3
+
+  constructor (providers) {
+    super()
+
+    if (typeof providers === 'string') {
+      providers = providers.split(',')
+    } else if (!Array.isArray(providers)) {
+      providers = [providers]
+    }
+
+    providers = providers.map(provider => (new Web3(provider)).eth.currentProvider)
```

Comment:
this is basically to turn each endpoint string into an instance of HttpProvider right? instead of going through this why not cast directly to httpprovider

Reply:
The list of providers can be comma separated strings, an array of strings or an array of providers.

This block of code ensures that we can support all of these without issue.

Reply:
ah right, makes sense, can you throw a comment to that effect